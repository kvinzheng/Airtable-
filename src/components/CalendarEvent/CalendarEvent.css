.CalendarEvent {
  border: 1px rgb(3, 155, 229) solid;
  color: white;
  border-radius: 3px;
  padding: 3px 5px;
  margin: 1px 3px;
}

.CalendarEvent-span {
  margin-right: 4px;
  font-weight: bold;
  font-size: 12px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 50px;
  width: 50px;
}
