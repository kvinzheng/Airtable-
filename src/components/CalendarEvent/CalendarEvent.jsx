import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import { setEvents, setEditableEvent } from "../../actions/events";
import { setCalendarForm } from "../../actions/form";

import "./CalendarEvent.css";

export const CalendarEvent = ({
  events,
  event,
  setEditableEvent,
  setCalendarForm,
}) => {

  const handleFormLoad = () => {
    const target = events.find((item) => item.id === event.id);

    if (target) {
      setCalendarForm(true);
      setEditableEvent({ ...target });
    }
  };
  return (
    <div
      className="CalendarEvent"
      data-testid="CalendarEvent-bar"
      onClick={handleFormLoad}
      style={{ backgroundColor: event.color }}
    >
      <span className="CalendarEvent-span">Title: {event.title}</span>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    events: state.events.data,
    form: state.form.calendarForm,
  };
};

CalendarEvent.propTypes = {
  events: PropTypes.array,
  form: PropTypes.bool,
};

export default connect(mapStateToProps, {
  setEvents,
  setEditableEvent,
  setCalendarForm,
})(CalendarEvent);
