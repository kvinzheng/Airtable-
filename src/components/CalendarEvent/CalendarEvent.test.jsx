import React from "react";

import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { CalendarEvent } from "./CalendarEvent";

import MockData from "../../helpers/testingData";

import { render, screen, fireEvent } from "@testing-library/react";

configure({ adapter: new Adapter() });


describe("CalendarEvent", () => {
  it("should render with data defined", () => {
    const component = shallow(
      <CalendarEvent
        setCalendarForm={jest.fn()}
        setEditableEvent={jest.fn()}
        events={MockData.events.data}
        form={MockData.form.calendarForm}
        event={{ color: "yellow", id: 1, title: "drink water" }}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("When you click on an event", () => {
    const setEditableEvent = jest.fn();
    const setCalendarForm = jest.fn();
    render(
      <CalendarEvent
        events={MockData.events.data}
        form={MockData.form.calendarForm}
        setEditableEvent={setEditableEvent}
        setCalendarForm={setCalendarForm}
        event={{ color: "yellow", id: 1, title: "drink water" }}
      />
    );
    fireEvent.click(screen.getAllByTestId("CalendarEvent-bar")[0]);

    expect(setEditableEvent).toHaveBeenCalled();
    expect(setCalendarForm).toHaveBeenCalled();
  });
});
