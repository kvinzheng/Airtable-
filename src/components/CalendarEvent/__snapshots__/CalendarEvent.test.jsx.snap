// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarEvent should render with data defined 1`] = `ShallowWrapper {}`;
